main.go

```diff
@@ -334,6 +382,40 @@ func runProtoc() int {
 	return 0
 }
 
+func download(url string) ([]byte, error) {
+	res, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	return ioutil.ReadAll(res.Body)
+}
+
+// downloadFile downloads and saves file
+func downloadFile(filepath string, url string) (err error) {
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("bad status: %s", resp.Status)
+	}
+
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		return err
```

Comment:
if the download failed for some reason, should we delete the partial file? will the application be able to recover without deleting the cache directory manually?

Reply:
I added handler for this situation to downloadProtoc() , it's in defer statement